crypto/rand/drbg_lib.c

```diff
@@ -761,7 +769,8 @@ int RAND_DRBG_set_callbacks(RAND_DRBG *drbg,
                             RAND_DRBG_get_nonce_fn get_nonce,
                             RAND_DRBG_cleanup_nonce_fn cleanup_nonce)
 {
-    if (drbg->state != DRBG_UNINITIALISED)
+    if (drbg->state != DRBG_UNINITIALISED
+            || drbg->parent != NULL)
```

Comment:
Prohibiting `RAND_DRBG_set_callbacks()` for child DRBGs is a severe restriction of the API and goes down the wrong path IMO :-/

Reply:
Sorry, but I do not see, how the callback can do the magic that rand_drbg_get_entropy
does when drbg->master != NULL.
How is it supposed to do "rand_drbg_lock(drbg->parent)"?

Reply:
Oh you are right! We did so many changes to the DRBG code in the last year that I wasn't quite aware anymore that the chaining was kept entirely private. And thinking twice about all these problems we will probably have an easier live if we never make this feature public. In other words: just forget my previous comment. :-)